pkg/operator/k8sutil/pvc_test.go

```diff
@@ -31,25 +32,48 @@ import (
 func TestExpandPVCIfRequired(t *testing.T) {
 
 	testcases := []struct {
-		label          string
-		currentPVCSize string
-		desiredPVCSize string
+		label            string
+		currentPVCSize   string
+		desiredPVCSize   string
+		expansionAllowed bool
 	}{
 		{
-			label:          "case 1: size is equal",
-			currentPVCSize: "1Mi",
-			desiredPVCSize: "1Mi",
+			label:            "case 1: size is equal",
+			currentPVCSize:   "1Mi",
+			desiredPVCSize:   "1Mi",
+			expansionAllowed: true,
 		},
 		{
-			label:          "case 2: current size is less",
-			currentPVCSize: "1Mi",
-			desiredPVCSize: "2Mi",
+			label:            "case 2: current size is less",
+			currentPVCSize:   "1Mi",
+			desiredPVCSize:   "2Mi",
+			expansionAllowed: true,
 		},
 		{
-			label:          "case 3: current size is more",
-			currentPVCSize: "2Mi",
-			desiredPVCSize: "1Mi",
+			label:            "case 3: current size is more",
+			currentPVCSize:   "2Mi",
+			desiredPVCSize:   "1Mi",
+			expansionAllowed: true,
 		},
+		{
+			label:            "case 4: storage class allow expand",
+			currentPVCSize:   "1Mi",
+			desiredPVCSize:   "2Mi",
+			expansionAllowed: true,
+		},
+		{
+			label:            "case 3: storage class does not allow expand",
+			currentPVCSize:   "1Mi",
+			desiredPVCSize:   "2Mi",
+			expansionAllowed: false,
+		},
+	}
+
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		AllowVolumeExpansion: func() *bool { flag := true; return &flag }(),
```

Comment:
we can get  rid of the anonymous function if `AllowVolumeExpansion` is always going to be `true`

Reply:
Done